homeassistant/components/stt/legacy.py

```diff
@@ -38,7 +38,7 @@ def async_get_provider(
 ) -> Provider | None:
     """Return provider."""
     if domain:
-        return hass.data[DATA_PROVIDERS].get(domain)
+        return cast(Provider | None, hass.data[DATA_PROVIDERS].get(domain))
 
     provider = async_default_provider(hass)
     return hass.data[DATA_PROVIDERS][provider] if provider is not None else None
```

Comment:
```suggestion
    providers: dict[str, Provider] = hass.data[DATA_PROVIDERS]
    if domain:
        return providers.get(domain)

    provider = async_default_provider(hass)
    return providers[provider] if provider is not None else None
```
If we assign `hass.data[DATA_PROVIDERS]` to a variable, we can add the type annotation without using `cast`. Added benefit being that we can reuse it in the second return statement.